File: two_sum.js
// Solution to: https://leetcode.com/problems/two-sum/

const twoSum = function(nums, target) {
  const output = {}
  for (let i = 0; i < nums.length; i++) {
      if (output[nums[i]] >= 0) {
          return [output[nums[i]] , i]
      }
      output[target-nums[i]] = i
  }
};

File: two_sum.rb
# Solution to: https://leetcode.com/problems/two-sum/

def two_sum(nums, target)
  dict = Hash.new
  nums.each_with_index do |num, i|
    if dict[target - num]
      return [dict[target - num], i]
    end
    dict[num] = i
  end
end
